objects/client.c

```diff
@@ -723,12 +723,58 @@
  * identifier remain the same. This allow to match a spawn event to an actual
  * client.
  *
+ * This is used to display a different mouse cursor when the application is
+ * loading and also to attach some properties to the newly created client (like
+ * a `tag` or `floating` state).
+ *
+ * Some applications, like `xterm`, don't support startup notification. While
+ * not perfect, the adding the following code to `rc.lua` will mitigate the
+ * issue. Please note that this code is Linux specific.
+ *
+ *    local blacklisted_snid = setmetatable({}, {__mode = "v" })
```

Comment:
I agree that some way for this not to grow unboundedly is needed, but why does tying this to the speed of the garbage collector make sense? However, I do not really have a better suggestion for this as we cannot know when the last process with the environment variable exits.